src/components/wizard/wizard.tsx

```diff
@@ -1,24 +1,24 @@
 import {
-  Button,
   Box,
+  Button,
   Card,
-  UnorderedList,
-  ListItem,
-  Text,
   Flex,
+  Text
 } from "@chakra-ui/react";
 import type { ReactNode } from "react";
-import { useState, Children } from "react";
-import { Form } from "react-final-form";
+import { Children, useState } from "react";
+import type { FieldValues, UseFormProps } from "react-hook-form";
+import { FormProvider, useForm } from "react-hook-form";
 
-// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-parameters
-export function Wizard<SchemaType extends object>({
-  onSubmit,
-  children
-}: {
+type WizardProps<SchemaType extends FieldValues> = {
   onSubmit: (v: SchemaType) => void;
   children: ReactNode;
-}) {
+  formProps?: UseFormProps<SchemaType>;
+};
+
+export function Wizard<SchemaType extends object>(props: WizardProps<SchemaType>) {
```

Comment:
this looks wrong then. As SchemaType here doesn't extends FieldValues. Should not even compile...